src/extension/wps/src/main/java/au/org/emii/aggregator/variable/UnpackerOverrides.java

```diff
@@ -0,0 +1,114 @@
+package au.org.emii.aggregator.variable;
+
+import ucar.ma2.DataType;
+
+import java.util.Arrays;
+
+/**
+ * Optional overrides for values to be used in the unpacked variable
+ */
+public class UnpackerOverrides {
+    private final Number newFillerValue;
+    private final Number[] newValidRange;
+    private final Number newValidMin;
+    private final Number newValidMax;
+    private final Number[] newMissingValues;
+    private final DataType newDataType;
+
+    private UnpackerOverrides(Builder builder) {
+        this.newFillerValue = builder.newFillerValue;
+        this.newValidRange = builder.newValidRange == null ? null :
+            Arrays.copyOf(builder.newValidRange, builder.newValidRange.length);
+        this.newValidMin = builder.newValidMin;
+        this.newValidMax = builder.newValidMax;
+        this.newMissingValues = builder.newMissingValues == null ? null :
+            Arrays.copyOf(builder.newMissingValues, builder.newMissingValues.length);
+        this.newDataType = builder.newDataType;
+    }
+
+    public Number getNewFillerValue() {
```

Comment:
There's a lot of type erasure here - using the Number superclass

- Are the encoding algorithms going to be allocating (eg. calling new) for every data value that has to be processed? GC will obviously handle it, but in terms of performance it might be better to pre-allocate once, and then reuse the objects.

- It's a pity Java's generic support isn't very strong, other wise you could parametize the encoders by netcdf data type (short, int, float, double), and handle things that way.